distributed/client.py

```diff
@@ -1732,14 +1745,18 @@ async def _():
     async def _shutdown(self):
         logger.info("Shutting down scheduler from Client")
 
+        self.status = "closing"
         for pc in self._periodic_callbacks.values():
             pc.stop()
-        if self.cluster:
-            await self.cluster.close()
-        else:
-            with suppress(CommClosedError):
-                self.status = "closing"
-                await self.scheduler.terminate()
+
+        async with self._wait_for_handle_report_task():
+            if self.cluster:
+                await self.cluster.close()
+            else:
+                with suppress(CommClosedError):
+                    await self.scheduler.terminate()
+
+        await self._close()
```

Comment:
Note: Closing the client is an extra cleanup step (i.e. not needed to make the user's client session quiet). However, it seemed strange that we didn't close it when the cluster was closed. Happy to rollback is it causes issues or folks would rather keep it open for some reason